src/table/components/TableBodyWrapper.jsx

```diff
@@ -27,23 +29,15 @@ const useStyles = makeStyles({
   },
 });
 
-function TableBodyWrapper({
-  rootElement,
-  tableData,
-  constraints,
-  selectionsAPI,
-  layout,
-  theme,
-  setShouldRefocus,
-  setFocusedCellCoord,
-  keyboard,
-  tableWrapperRef,
-  announce,
-}) {
+function TableBodyWrapper({ setShouldRefocus, setFocusedCellCoord, tableWrapperRef }) {
+  const {
+    nebulaProps: { rootElement, layout, tableData, constraints, selectionsAPI, keyboard, theme },
+  } = useRootContext();
   const { rows, columns } = tableData;
   const hoverEffect = layout.components?.[0]?.content?.hoverEffect;
   const bodyStyle = useMemo(() => getBodyStyle(layout, theme), [layout, theme.name()]);
   const classes = useStyles(bodyStyle);
+  const announce = useAnnounce();
```

Comment:
This is great and all, but I'm curious what it looks like when we have tackled a bunch of the utility functions. Now we get `announce` from `useAnnounce`, but we can only fetch it like this in a file using react. Later in this file we pass it to `bodyHandleKeyPress`. Is there a way of getting around that? defining `bodyHandleKeyPress` somewhere so you can run `useAnnounce` to get it, so we don't have to pass it?

Reply:
Yes, absolutely

later when we are dealing with useA11y hook, we will initiate our factory about announcement inside of `useA11y` hook and then provide it thought the context, which will be available to use thought the entire project.

in case of util functions like `bodyHandleKeyPress` as much as we can, we will move the functionality inside of the hook, but the general rule still will be there for util functions. this means that util functions will still kept as utilities  as much as we agree on their functionality.